edge/pkg/servicebus/servicebus.go

```diff
@@ -88,72 +121,154 @@ func processMessage(msg *beehiveModel.Message) {
 		return
 	}
 	resource := msg.GetResource()
-	r := strings.Split(resource, ":")
-	if len(r) != 2 {
-		m := "the format of resource " + resource + " is incorrect"
-		klog.Warningf(m)
-		code := http.StatusBadRequest
-		if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
-			beehiveContext.SendToGroup(modules.HubGroup, response)
+	switch msg.GetOperation() {
+	case "start":
+		if atomic.CompareAndSwapInt32(&inited, 0, 1) {
+			dao.InsertUrls(resource)
+			go server(c)
 		}
-		return
-	}
-	content, err := json.Marshal(msg.GetContent())
-	if err != nil {
-		klog.Errorf("marshall message content failed %v", err)
-		m := "error to marshal request msg content"
-		code := http.StatusBadRequest
-		if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
-			beehiveContext.SendToGroup(modules.HubGroup, response)
+	case "stop":
+		dao.DeleteUrlsByKey(resource)
+		c <- struct{}{}
```

Comment:
and we can only process one URL one time?

Reply:
yeap,that's my fault. We should judge if the table target_urls is empty，if true we will send stop to the channel for stopping the servicebus server